### PR TITLE
[docs] Add docs for `flake8-implicit-str-concat` rules

### PR DESCRIPTION
**Pull request by thatlittleboy**
Added docs for the ISC rules.

Related: https://github.com/charliermarsh/ruff/issues/2646

### CODE REVIEW

File: crates/ruff/src/rules/flake8_implicit_str_concat/rules.rs [hunk 1]
```diff
@@ -19,6 +36,33 @@ impl Violation for SingleLineImplicitStringConcatenation {
...
+    /// Use instead:
+    /// ```python
+    /// z = (
+    ///     "The quick brown fox jumps over the lazy "
+    ///     "dog."
+    /// )
+    ///
+    /// ## References
+    /// * [PEP 8](https://peps.python.org/pep-0008/#maximum-line-length)
+    /// ```
```
Comment (sbrugman) [inline crates/ruff/src/rules/flake8_implicit_str_concat/rules.rs:65]: Code block brackets should be placed just after the example (now include references)

Comment (thatlittleboy) [reply crates/ruff/src/rules/flake8_implicit_str_concat/rules.rs:65]: Sorry Not quite sure I follow, can you illustrate what you mean?

Comment (sbrugman) [reply crates/ruff/src/rules/flake8_implicit_str_concat/rules.rs:65]: The three backticks (```) need to be moved up a couple of lines.

Comment (thatlittleboy) [reply crates/ruff/src/rules/flake8_implicit_str_concat/rules.rs:65]: oh yea, nice catch, thanks!

**Pull request merged**